apps/src/templates/BaseDialog.jsx

```diff
@@ -0,0 +1,148 @@
+// TODO: Temporary measure to make sure code-studio doesn't end up with two Reacts
+var React = window.React;
+
+/**
+ * BaseDialog
+ * A generic modal dialog that has an x-close in the upper right, and a
+ * semi-transparent backdrop. Can be closed by clicking the x, clicking the
+ * backdrop, or pressing esc.
+ */
+var BaseDialog = React.createClass({
+  propTypes: {
+    isOpen: React.PropTypes.bool,
+    handleClose: React.PropTypes.func,
+    uncloseable: React.PropTypes.bool,
+    hideBackdrop: React.PropTypes.bool,
+    useDeprecatedGlobalStyles: React.PropTypes.bool,
+  },
+
+  componentDidMount: function () {
+    this.focusDialog();
+  },
+
+  componentDidUpdate: function () {
+    this.focusDialog();
+  },
+
+  closeOnEscape: function (event) {
+    if (event.key === 'Escape') {
+      this.closeDialog();
+    }
+  },
+
+  closeDialog: function () {
+    if (!this.props.uncloseable && this.props.handleClose) {
+      this.props.handleClose();
+    }
+  },
+
+  focusDialog: function () {
+    if (this.props.isOpen) {
+      this.refs.dialog.focus();
+    }
+  },
+
+  render: function () {
+    if (!this.props.isOpen && !this.props.hideBackdrop) {
+      return <div></div>;
+    }
+
+    var bodyStyle;
+    if (this.props.hideBackdrop) {
+      bodyStyle = {
+        position: 'initial',
+        marginLeft: 0,
+      };
+    }
+
+    let modalClassNames = "modal";
+    let modalBodyClassNames = "modal-body";
+    let modalBackdropClassNames = "modal-backdrop";
+
+    if (this.props.useDeprecatedGlobalStyles) {
+      modalClassNames = "modal dash_modal in";
+      modalBodyClassNames = "modal-body dash_modal_body";
+      modalBackdropClassNames = "modal-backdrop in";
+    }
```

Comment:
The deprecated styles I'm referring to here are the ones related to the `dash_modal` css class.

A question that I'd like people's opinion on is whether or not I should also ditch using the `modal`, `modal-body`, and `modal-backdrop` css classes as well. To do so would mean copying the styles from the scss files into javascript. I'm not sure where else we are using these css classes so I would be reluctant to get rid of them outright. But at the same time, copying them will mean keeping two things in sync.

Perhaps @breville knows where else we are using these css classes and hopefully he has an idea about how destructive just deleting them would be?